online-shop/src/app/product-details/product-details.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Output, EventEmitter } from '@angular/core';
+import { PRODUCTS } from '../products';
+import { Product } from '../product';
+
+@Component({
+  selector: 'app-product-details',
+  templateUrl: './product-details.component.html',
+  styleUrls: ['./product-details.component.css']
+})
+
+export class ProductDetailsComponent implements OnInit {
+
+  product;
```

Comment:
If this is a dumb component, then this field must be an input (and the find stuff should happen inside the smart component). Also, always give types to your fields (`product: Product`).